tests/dragonfly/utility.py

```diff
@@ -432,18 +433,22 @@ def should_run():
 
     async def _executor_task(self, db, queue):
         client = aioredis.Redis(port=self.port, db=db)
+        command_count = 0
         while True:
             cmds = await queue.get()
             if cmds is None:
                 queue.task_done()
                 break
-
-            pipe = client.pipeline(transaction=False)
+            is_multi_transaction = False
+            if command_count == self.multi_transaction_ratio:
+                is_multi_transaction = True
+            pipe = client.pipeline(transaction=is_multi_transaction)
             for cmd in cmds:
                 pipe.execute_command(cmd)
 
             await pipe.execute()
             queue.task_done()
+            ++command_count
         await client.connection_pool.disconnect()
```

Comment:
Adi, there is no pre-increment in Python 😄 

Besides, the whole flow above is wrong. Those are batches generated by the generator - with a default size of 1000 there are usually just a few on the tests.  I've seen you reduced the size, so there are more, probably a few dozens. If you run the tests with `-s` it prints how many batches were submitted.

That doesn't include the runs with a target number of batches, i.e. target_times. Lets rename it to target_ops then and make it calculate the number of batches. If you want, I can do it quickly.

Now you could would run this only one if command_count whould've ever reached self.multi_transaction_ratio. Instead, we can do it probabilistic, i.e. if run it if `random.random() < transaction_prob`

Also one more small nit, in the snapshot_tests, set it to a zero value, so they don't spend time on filling it with tx

Reply:
One more issue, lets encode this on the producer, inside `_generator_task`, so it can print in in the file as well, where one tx started and where it ended

Reply:
I dont understand the comment about set it to 0 in snapshot_tests?  what is "they" and "it" when you say "so they don't spend time on filling it with tx"
Isnt it good to test the multi flow in the snapshot tests as well?

Reply:
Also did not understand this:
That doesn't include the runs with a target number of batches, i.e. target_times. Lets rename it to target_ops then and make it calculate the number of batches.

From what I understand when calling run with target_times, we call the client.pipeline  target_times times. Each call will execute batch_size commands (default 100).
What I want to do now is with probability of multi_transaction_probability make the call to client.pipeline with transaction=true. This will execute this batch_size commands  in one multi transaction.
This is the logic in the PR, I just commited a fix. are you suggesting something else?